crypto/evp/ctrl_params_translate.c

```diff
@@ -1008,6 +1008,38 @@ static int fix_dh_nid(enum state state,
     return ret;
 }
 
+/* EVP_PKEY_CTRL_DH_RFC5114 */
+static int fix_dh_nid5114(enum state state,
+                          const struct translation_st *translation,
+                          struct translation_ctx_st *ctx)
+{
+    int ret;
+
+    if ((ret = default_check(state, translation, ctx)) <= 0)
+        return ret;
+
+    /* This is currently only settable */
+    if (ctx->action_type != SET)
+        return 0;
+
+    if (state == PRE_CTRL_STR_TO_PARAMS) {
```

Comment:
@levitte - Have I done this right?? It seems there are no other case of this PRE_CTRL_STR_TO_PARAMS being used like this, so either I am doing something wrong or there is possible wrongness with other entries.

Reply:
IMO we do not have any other CTRLs where you convert an integer value into a named parameter.